enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -529,6 +529,15 @@ def resolve_policy(cls, redeemable_policies):
         )
         return sorted_policies[0]
 
+    def delete(self, *args, **kwargs):
+        """
+        Perform a soft-delete, overriding the standard delete() method to prevent hard-deletes.
+
+        If this instance was already soft-deleted, invoking delete() is a no-op.
+        """
+        self.active = False
```

Comment:
nit: you could set the reason in here from kwargs if you wanted.

Reply:
oh nice, I wasn't sure if changing the call signature of delete was kosher